test/common.py

```diff
@@ -119,13 +119,22 @@ def assertEqual(self, x, y, prec=None, message=''):
 
         if torch.is_tensor(x) and torch.is_tensor(y):
             def assertTensorsEqual(a, b):
-                max_err = 0
                 super(TestCase, self).assertEqual(a.size(), b.size())
-                for index in iter_indices(a):
-                    max_err = max(max_err, abs(a[index] - b[index]))
-                self.assertLessEqual(max_err, prec, message)
+                if a.numel() > 0:
```

Comment:
good point. I checked the various operations, most of them are already covered (for example `max` doesn't use tensor comparisons), however I fixed `abs` to use the slow comparison.